aiven/resource_vpc_peering_connection.go

```diff
@@ -286,33 +287,90 @@ func resourceVPCPeeringConnectionRead(_ context.Context, d *schema.ResourceData,
 	return copyVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(d, pc, projectName, vpcID)
 }
 
-func resourceVPCPeeringConnectionDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(*aiven.Client)
 
 	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+
 	isAzure, err := isAzureVPCPeeringConnection(d, client)
 	if err != nil {
 		return diag.Errorf("Error checking if it Azure VPC peering connection: %s", err)
 	}
 
 	if isAzure {
 		if peerResourceGroup, ok := d.GetOk("peer_resource_group"); ok {
-			err := client.VPCPeeringConnections.DeleteVPCPeeringWithResourceGroup(
-				projectName, vpcID, peerCloudAccount, peerVPC, peerResourceGroup.(string), peerRegion)
-			if err != nil {
+			if err = client.VPCPeeringConnections.DeleteVPCPeeringWithResourceGroup(
+				projectName,
+				vpcID,
+				peerCloudAccount,
+				peerVPC,
+				peerResourceGroup.(string),
+				peerRegion,
+			); err != nil && !aiven.IsNotFound(err) {
 				return diag.Errorf("Error deleting VPC peering connection with resource group: %s", err)
 			}
-			return nil
 		} else {
 			return diag.Errorf("cannot delete an Azure VPC peering connection without `peer_resource_group`")
 		}
 	}
-
-	err = client.VPCPeeringConnections.DeleteVPCPeering(projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
-	if err != nil && !aiven.IsNotFound(err) {
+	if err = client.VPCPeeringConnections.DeleteVPCPeering(
+		projectName,
+		vpcID,
+		peerCloudAccount,
+		peerVPC,
+		peerRegion,
+	); err != nil && !aiven.IsNotFound(err) {
 		return diag.Errorf("Error deleting VPC peering connection: %s", err)
 	}
 
+	if _, err := (&resource.StateChangeConf{
+		Pending: []string{
+			"ACTIVE",
+			"APPROVED",
+			"APPROVED_PEER_REQUESTED",
+			"DELETING",
+			"INVALID_SPECIFICATION",
+			"PENDING_PEER",
+			"REJECTED_BY_PEER",
+		},
+		Target: []string{
+			"DELETED",
+			"DELETED_BY_PEER",
```

Comment:
I think `DELETED_BY_PEER` should be in the Pending states

Reply:
`DELETED_BY_PEER` means: The customer deleted the peering connection through their own cloud account. That is, the customer deleted the peering cloud resource in their account. Executor detects this when regularly rebuilding existing peering connections in the `PENDING_PEER` and `ACTIVE` states, and noticed that the connection cloud resource is in a deleted state or is missing and we infer the customer must have done something about it. There are no transitions from this state

Reply:
done